install-deps.sh

```diff
@@ -322,6 +322,7 @@ else
                 $with_zbd && install_libzbd_on_ubuntu bionic
                 ;;
             *Focal*)
+                ensure_decent_gcc_on_ubuntu 11 focal
```

Comment:
@cbodley @adamemerson i'd suggest at revise this commit , so it only installs gcc-11 without pointing gcc to gcc-11. to avoid being bitten by https://tracker.ceph.com/issues/54455

Reply:
because libc++ is not packaged by llvm-toolset packages for el8. i am proposing following combinations

- for running unit test: 
  - clang-12 + libstdc++11  for verifying pull requests for more picky compiler (most of the time), better error messages, less compile-time memory footprint and better compile warnings.
- for building rpm / deb packages 
  - clang-12 + libstdc++11 for using exactly the same combination with the one used by PR for less surprising after pushing changes to ceph-ci
  - gcc-11 + libstdc++11 should always work. but *might* fail to build as the combination is not verified by the jenkins job verifying pull requests

currently, `run-make-check.sh` is used for running make check, it tries to use clang if it's available, and clang detects the libstdc++ use some heuristics as you've noticed. when it comes to `pbuilder` and `mock` (or `rpmbuild`, we have to teach them by pointing `CMAKE_CXX_COMPILER` to `clang++-12` or use something like`scl enable gcc-toolset-11`.